cpp/src/aztec/honk/composer/composer_helper/permutation_helper.hpp

```diff
@@ -94,7 +97,9 @@ void compute_wire_copy_cycles(CircuitConstructor& circuit_constructor, CycleColl
         wire_copy_cycles[static_cast<size_t>(w_3_index)].emplace_back(static_cast<uint32_t>(i + num_public_inputs),
                                                                       WireType::OUTPUT);
 
+        // Handle width 4 separately
         if constexpr (program_width > 3) {
+            static_assert(program_width == 4);
             const auto w_4_index = real_variable_index[w_4[i]];
             if (w_4_index >= number_of_cycles) {
                 wire_copy_cycles.resize(w_4_index + 1);
```

Comment:
I agree. This is probably the most inefficient way to do it, too, since by just starting from the end, I'd quickly get the actual maximum value. I'll delete it. Thanks